example-extension/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+use std::marker::PhantomData;
+use std::os::raw::{c_char, c_int};
+
+use rusqlite::vtab::{
+    eponymous_only_module, sqlite3_vtab, sqlite3_vtab_cursor, Context, IndexInfo, VTab,
+    VTabConnection, VTabCursor, Values,
+};
+use rusqlite::{
+    ffi,
+    functions::FunctionFlags,
+    types::{ToSqlOutput, Value},
+};
+use rusqlite::{to_sqlite_error, Connection, Result};
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+pub extern "C" fn sqlite3_extension_init(
```

Comment:
Commented on by @thomcc in #631 https://github.com/rusqlite/rusqlite/pull/631#discussion_r404001494
"
This probably should be `unsafe fn` though. It both writes to a static mut without synchronization and dereferences the raw pointers passed in as arguments.
"

Reply:
fixed